lib/src/repo.rs

```diff
@@ -625,27 +658,10 @@ impl MutableRepo {
         RepoRef::Mutable(self)
     }
 
-    pub fn base_repo(&self) -> &Arc<ReadonlyRepo> {
+    pub fn readonly_repo(&self) -> &Arc<ReadonlyRepo> {
```

Comment:
Just a comment. We could instead implement `Repo` for `Arc<ReadonlyRepo>`, and require `base_repo() -> &Arc<...>`. However, most callers don't need `&Arc`, so it wouldn't be worth.

Reply:
Good point. I had not even considered that. Yes, most callers don't need `&Arc`, but almost all of them already have it, so maybe it's worth it after all? I'll add a commit on top to show what it looks like. I like. If you agree, I'll clean up the series.

Reply:
Indeed, `Arc<ReadonlyRepo>` is the data type we use everywhere, so new version looks good to me.

Reply:
I squashed in most of that last commit, but also added a tiny new commit early in the series. Feel free to take another look and mark this resolved (I'll enable auto-merge).

Reply:
LGTM, thanks.